Updated provides and autoloads for compatibility with use-package lazy loading

cormacc opened the pull request:
Hi,

Thanks for this effort - I was using a slightly modified version of the old doxymacs.el, but this is much nicer.

Preparing a PR to add this to the spacemacs c/c++ layer and made the following changes...

Replaced all the per-function (provide ..) with just the package name ~(provide 'gendoxy)~, as this is consistent with the pattern in other el packages I've looked at / use-package expectations (it spits out an error otherwise).

Added ;;;###autoload annotation to the user-facing functions to support lazy-loading.

Regards,
C

mp81ss commented:
You changed the newline format and I cannot see the changes in github.
Please fix it

cormacc commented:
Apologies. Had set "text=..." incorrectly in .gitattributes - amended and force-pushed to my fork. Github weirdly seems to have closed the PR as a result though... I'll open a new PR

cormacc committed:
Minor amendments for use-package compatibility.

Added ';;;###autoloads' annotations to interactive function definitions.
Switched from multiple to single 'provide' as per package conventions.

Added .gitattributes to handle cross-platform line-endings gracefully.

cormacc commented:
... and github appears to have come to its senses and let me reopen this one.
